lib/adyen/rest/client.rb

```diff
@@ -68,6 +71,7 @@ def http
       #   of  executing the underlying HTTP request.
       def execute_request(request)
         request.validate!
+        set_endpoint(request.resource)
```

Comment:
If the endpoint is potentially different for every request, there's not really a point in saving it as a instance variable anymore.

Maybe make `endpoint` part of the Request class instead?